src/dmd/root/outbuffer.d

```diff
@@ -379,13 +380,29 @@ struct OutBuffer
      * Params:
      *  u = integral value to append
      */
-    extern (C++) void print(ulong u) pure nothrow
+    extern (C++) void print(uint64_t u) pure nothrow
```

Comment:
Why the change ?

Reply:
I've added a C++ usage of it, so I'm following the @ibuclaw [advise](https://github.com/dlang/dmd/pull/11555#discussion_r469192870).

Reply:
Don't think that'd work on OSX.  You need a new `d_ulong` type similar to `d_size_t` that checks the size of `long`.

Reply:
Though I don't recall for how long the cpp_long fix has been in druntime for, but I think not long enough to fall within the period when dmd was broken for a few releases.

Reply:
Oh man, why something that simple is not yet handled easily.